### PR TITLE
Fix stop and status logic 

### LINKED ISSUE #3795: [mesheryctl] Myopic logic in `system status` and `system stop`
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
```
➜  Meshery mesheryctl system stop
Meshery is not running. Nothing to stop.
➜  Meshery mesheryctl system status
Meshery is not running. Run `mesheryctl system start` to start Meshery.
➜  Meshery kubectl get pods -n meshery
NAME                                    READY   STATUS    RESTARTS   AGE
meshery-broker-0                        2/2     Running   0          2m7s
meshery-consul-dcdb77f96-wvjtp          1/1     Running   0          3m47s
meshery-cpx-79fb9656bc-4rcbd            1/1     Running   0          3m47s
meshery-istio-64c757bbc8-hhcp6          1/1     Running   0          3m47s
meshery-kuma-6669cc5979-g92wh           1/1     Running   0          3m47s
meshery-linkerd-5c8c7756db-2hhss        1/1     Running   0          3m47s
meshery-meshsync-7cc779c8bb-rtmpd       1/1     Running   5          3m27s
meshery-nsm-78b44df4f6-lljgs            1/1     Running   0          3m47s
meshery-operator-bddf9fd66-ndqbn        2/2     Running   0          3m47s
meshery-osm-7bb65bf558-z9txc            1/1     Running   0          3m46s
meshery-traefik-mesh-6778958fd4-qt865   1/1     Running   0          3m46s
```

#### Desired Behavior
These two commands should not only take `meshery` into consideration but all components listed in the `current-context`.

- Host OS: Mac
- Meshery Server Version: stable-v0.5.42
- Meshery Client Version: stable-v0.5.42
- Platform: Kubernetes

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: metonymic-smokey <ahuja.aditi@gmail.com>

**Description**

This PR fixes #3795 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by metonymic-smokey** [comment]
Current status: For both platforms, the meshery namespace is checked and pods are stopped. Earlier, this was done only in the case of kubernetes.

**Comment by metonymic-smokey** [comment]
@piyushsingariya the integration tests for restart are failing here, because some manifest file is missing. Is it because I'm not fetching manifests in `stop` and it is assumed there are no manifests?
Start and stop aren't failing though.

**Comment by piyushsingariya** [comment]
> @piyushsingariya the integration tests for restart are failing here, because some manifest file is missing. Is it because I'm not fetching manifests in `stop` and it is assumed there are no manifests?
> Start and stop aren't failing though.

yes it's probably because of that as restart is nothing but stop() and then start().

**Comment by metonymic-smokey** [comment]
ah okay, thanks!
Then this failing test will probably get fixed only when #3897 is fixed.

**Comment by navendu-pottekkat** [comment]
@piyushsingariya @metonymic-smokey I had rerun the workflows but still the test fails, could you check?

**Comment by piyushsingariya** [comment]
#3897 needs to be fixed for the workflow check!

File: mesheryctl/internal/cli/root/system/stop.go [hunk 1]
```diff
@@ -116,48 +126,36 @@ func stop() error {
...
+	hcOptions := &HealthCheckOptions{
+		PrintLogs:           false,
+		IsPreRunE:           false,
+		Subcommand:          "",
+		RunKubernetesChecks: true,
+	}
+	hc, err := NewHealthChecker(hcOptions)
+	if err != nil {
+		return ErrHealthCheckFailed(err)
+	}
```
Comment (piyushsingariya) [inline mesheryctl/internal/cli/root/system/stop.go:141]: will this func ever return an error? 🤔

**Comment by piyushsingariya** [comment]
Alarm! Integration tests are failing.

**Comment by metonymic-smokey** [comment]
Yep, the other PR needs to be merged for that.

**Comment by navendu-pottekkat** [comment]
@metonymic-smokey @piyushsingariya Is this PR ready for another review after the merge conflicts have been fixed?

**Comment by piyushsingariya** [comment]
@metonymic-smokey There's a merge conflict here.

**Comment by metonymic-smokey** [comment]
@piyushsingariya thanks! I'll fix this in a bit, sorry for the delay

**Pull request closed without merging**